src/include/platform/internal/GenericPlatformManagerImpl_FreeRTOS.ipp

```diff
@@ -241,25 +269,109 @@ CHIP_ERROR GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_StartEventLoopTask(v
 {
 #if defined(CHIP_CONFIG_FREERTOS_USE_STATIC_TASK) && CHIP_CONFIG_FREERTOS_USE_STATIC_TASK
     mEventLoopTask = xTaskCreateStatic(EventLoopTaskMain, CHIP_DEVICE_CONFIG_CHIP_TASK_NAME, ArraySize(mEventLoopStack), this,
-                                       CHIP_DEVICE_CONFIG_CHIP_TASK_PRIORITY, mEventLoopStack, &mventLoopTaskStruct);
+                                       CHIP_DEVICE_CONFIG_CHIP_TASK_PRIORITY, mEventLoopStack, &mEventLoopTaskStruct);
 #else
     xTaskCreate(EventLoopTaskMain, CHIP_DEVICE_CONFIG_CHIP_TASK_NAME, CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE / sizeof(StackType_t),
                 this, CHIP_DEVICE_CONFIG_CHIP_TASK_PRIORITY, &mEventLoopTask);
 #endif
-
     return (mEventLoopTask != NULL) ? CHIP_NO_ERROR : CHIP_ERROR_NO_MEMORY;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_StopEventLoopTask(void)
+{
+    mShouldRunEventLoop.store(false);
+    return CHIP_NO_ERROR;
+}
+
 template <class ImplClass>
 void GenericPlatformManagerImpl_FreeRTOS<ImplClass>::EventLoopTaskMain(void * arg)
 {
-    ChipLogDetail(DeviceLayer, "CHIP task running");
+    ChipLogDetail(DeviceLayer, "CHIP event task running");
     static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->Impl()->RunEventLoop();
     // TODO: At this point, should we not
     // vTaskDelete(static_cast<GenericPlatformManagerImpl_FreeRTOS<ImplClass> *>(arg)->mEventLoopTask)?
     // Or somehow get our caller to do it once this thread is joined?
 }
 
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_PostBackgroundEvent(const ChipDeviceEvent * event)
+{
+    if (mBackgroundEventQueue == NULL)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+    if (!(event->Type == DeviceEventType::kCallWorkFunct || event->Type == DeviceEventType::kNoOp))
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    auto status = xQueueSendToBack(mBackgroundEventQueue, event, 1);
+    if (status != pdTRUE)
+    {
+        ChipLogError(DeviceLayer, "Failed to post event to CHIP background event queue");
+        return CHIP_ERROR(chip::ChipError::Range::kOS, status);
+    }
```

Comment:
We should have CHIP_ERROR_RESOURCE_EXHAUSTED working/detected for queue full.

Reply:
Rather, in both cases the error range is OS and the value is whatever was returned from the FreeRTOS function. Are you saying we should just return resource-exhausted instead? I'm down with that, the FreeRTOS return values don't seem helpful.

Reply:
RESOURCE_EXHAUSTED is an IM error, so instead I used CHIP_ERROR_NO_MEMORY.

Reply:
I did not change the existing error code in PostEvent (the foreground version) or document that function; it's possible someone is checking its return and expecting the current error (kOS plus status from FreeRTOS function).